src/shared/test/keyboard-navigation-test.js

```diff
@@ -0,0 +1,304 @@
+import { options as preactOptions, render } from 'preact';
+import { useRef } from 'preact/hooks';
+import { act } from 'preact/test-utils';
+
+import { useArrowKeyNavigation } from '../keyboard-navigation';
+import { waitFor } from '../../test-util/wait';
+
+function Toolbar({ navigationOptions = {} }) {
+  const containerRef = useRef();
+
+  useArrowKeyNavigation(containerRef, navigationOptions);
+
+  return (
+    <div ref={containerRef} data-testid="toolbar">
+      <button data-testid="bold">Bold</button>
+      <button data-testid="italic">Italic</button>
+      <button data-testid="underline">Underline</button>
+      <a href="/help" target="_blank" data-testid="help">
+        Help
+      </a>
+    </div>
+  );
+}
+
+describe('shared/keyboard-navigation', () => {
+  describe('useArrowKeyNavigation', () => {
+    let container;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      document.body.append(container);
+      renderToolbar();
+    });
+
+    afterEach(() => {
+      container.remove();
+    });
+
+    // Workaround for an issue with `useEffect` throwing exceptions during
+    // `act` callbacks. Can be removed when https://github.com/preactjs/preact/pull/3530 is shipped.
+    let prevDebounceRendering;
+    beforeEach(() => {
+      prevDebounceRendering = preactOptions.debounceRendering;
+    });
+    afterEach(() => {
+      preactOptions.debounceRendering = prevDebounceRendering;
+    });
+
+    function renderToolbar(options = {}) {
+      // We render the component with Preact directly rather than using Enzyme
+      // for these tests. Since the `tabIndex` state lives only in the DOM,
+      // and there are no child components involved, this is more convenient.
+      act(() => {
+        render(<Toolbar navigationOptions={options} />, container);
+      });
+      return findElementByTestId('toolbar');
+    }
+
+    function findElementByTestId(testId) {
+      return container.querySelector(`[data-testid=${testId}]`);
+    }
+
+    function pressKey(key) {
+      const event = new KeyboardEvent('keydown', {
+        bubbles: true,
+        cancelable: true,
+        key,
+      });
+      act(() => {
+        findElementByTestId('toolbar').dispatchEvent(event);
+      });
+      return event;
+    }
+
+    function currentItem() {
```

Comment:
This sounds more like it would return an element, but returns a string. Not a big deal as it's just a test helper. I do wonder if it would be clearer not to abstract this but just check `document.activeElement.innerText` directly inline.

Reply:
Maybe `currentItemLabel` would be better? I added this helper to encapsulate the details of getting the label of the current element, as I think it changed a couple of times during development.